SDWebImage/Core/SDImageCache.m

```diff
@@ -191,6 +202,10 @@ - (void)storeImage:(nullable UIImage *)image
     // if memory cache is enabled
     if (toMemory && self.config.shouldCacheImagesInMemory) {
         NSUInteger cost = image.sd_memoryCost;
+        if (!image && imageData) {
+            cost = imageData.length;
+            image = [UIImage imageWithData:imageData];
```

Comment:
If animation image like webp I think this logic is wrong

Reply:
I think you can make things easier. For example:

What if people store an Image Data which can not been decoded ? Should this function return earily or just store the disk data.

+ If return earily, no data is stored to disk.
+ If can decoded, then the image must use the same API and logic when call `queryImageForKey:`, right ? So you just call the `SDImageCacheDecodeImageData`, which is used by third-party cache impl as we documentetd.